### PR TITLE
Init fix

### PR DESCRIPTION
**Pull request by nikolapeevski-mentormate**

### CODE REVIEW

File: src/android/CordovaBrightcoveCallbackUtil.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    public CallbackContext getCallbackContext() {
+        return callbackContext;
+    }
+
+    public void setCallbackContext(CallbackContext callbackContext) {
+        this.callbackContext = callbackContext;
+    }
+
+    private static class CordovaBrightcoveCallbackUtilHolder {
+        public final static CordovaBrightcoveCallbackUtil INSTANCE = new CordovaBrightcoveCallbackUtil();
```
Comment (joesbigidea) [inline src/android/CordovaBrightcoveCallbackUtil.java:25]: Why isn't INSTANCE just a field on CordovaBrightcoveCallbackUtil?

**Review by joesbigidea — CHANGES_REQUESTED** [review]
Looking good! I put a few comments/questions on here

File: src/android/BrightcoveActivity.java [hunk 2]
```diff
@@ -3,63 +3,126 @@
...
 
-        playById(brightcovePolicyKey, brightcoveAccountId, videoId);
-    }
+        if (intent.hasExtra("video-url")) {
+            this.videoUrl = intent.getStringExtra("video-url");
+            this.playByUrl();
+        } else {
+            this.brightcovePolicyKey = intent.getStringExtra("brightcove-policy-key");
+            this.brightcoveAccountId = intent.getStringExtra("brightcove-account-id");
+            this.videoId = intent.getStringExtra("video-id");
```
[the remark this answers is not in the document]
Comment (nikolapeevski-mentormate) [reply src/android/BrightcoveActivity.java:59]: Okay, I"ll address it shortly.

File: src/android/BrightcoveActivity.java [hunk 3]
```diff
@@ -3,63 +3,126 @@
...
+            jsonObject.put("percentage", 100);
+        } else {
+            jsonObject.put("percentage", this.getPlaybackPercentage());
+        }
+
+        return jsonObject;
+    }
+
+    private int getPlaybackPercentage() {
+        return (brightcoveVideoView.getCurrentPosition() * 100) / brightcoveVideoView.getDuration();
```
[the remark this answers is not in the document]
Comment (nikolapeevski-mentormate) [reply src/android/BrightcoveActivity.java:126]: It returns -1

[the remark this answers is not in the document]
Comment (nikolapeevski-mentormate) [reply src/android/BrightcoveActivity.java:126]: I've tried to close the video a few times and it returned percentage: 0.

**Review by joesbigidea — APPROVED** [review]

**Review by nsolter — APPROVED** [review]
This looks fine overall - I didn't review line-by-line, though.

**Pull request merged**